src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Repositories/OrganisationRepository.cs

```diff
@@ -68,50 +92,41 @@ public async Task<string> LookupOrgNumber(string org)
         {
             await HarvestOrgsIfCacheIsMoreThanOneHourOld();
 
-            Organisation organisation = orgToOrganisation.GetValueOrDefault(org, null);
+            string cacheKey = $"org-{org}";
+            Organisation organisation;
+
+            _memoryCache.TryGetValue(cacheKey, out organisation);
 
             return organisation?.OrgNumber;
         }
 
         private async Task HarvestOrgsIfCacheIsMoreThanOneHourOld()
         {
-            DateTime timestamp = DateTime.Now;
-            timestamp = timestamp.AddHours(-1);
-
-            if (dictionaryLastUpdated < timestamp || orgNumberToOrganisation.Count == 0)
+            if (_lastHarvest < DateTime.UtcNow.AddHours(-1))
```

Comment:
It is a theoretical possibility that one request will hit an empty cache. The window is tiny, but because `_lastHarvest` is set after all entries in the cache, then cache might be empty right before the timer. This can be fixed by setting the cache expiration to be longer than the offset in this if.